_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,367 @@
+
+The connector creation API, `/_plugins/_ml/connectors/_create`, creates connections to third-party ML tools. Using the `endpoint` parameter, you can connect ML Commons to any supported ML tool using its specific API endpoint. For example, to connect to a ChatGPT completion model, you can connect using the `api.openai.com`, as shown in the following example:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "OpenAI Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+        "endpoint": "api.openai.com",
+        "auth": "API_Key",
+        "content_type": "application/json",
+        "max_tokens": 7,
+        "temperature": 0,
+        "model": "text-davinci-003"
+    },
+    "credential": {
+        "openAI_key": "<encrypted-key>"
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "url": "https://${parameters.endpoint}/v1/completions",
+            "headers": {
+                "Authorization": "Bearer ${credential.openAI_key}"
+            },
+            "request_body": "{ \"model\": \"${parameters.model}\", \"prompt\": \"${parameters.prompt}\", \"max_tokens\": ${parameters.max_tokens}, \"temperature\": ${parameters.temperature} }"
+        }
+    ],
+    "access_mode": "public"
+}
+```
+
+If successful, the connector API responds with a `connector_id` and `status` for the connection:
+
+```
+{
+ "connector_id": "M2pBmYgB5cK1H70wk3hd",
+ "status": "CREATED"
+}
+```
+
+## Configuration options
+
+The following configuration options are **required** in order to create a connector:
+
+| Field | Data type | Description |
+| :---  | :--- | :--- |
+| `name` | String | The name of the connector. |
+| `description` | String | A description of the connector. |
+| `version` | Integer | The version of the connector. |
+| `protocol` | String | The protocol for the connection. For AWS services such as Amazon SageMaker and Amazon Bedrock, use `aws_sigv4`. For all other services, use `http`. |
+| `parameter` | JSON array | The default connector parameters, including `endpoint` and `model`. 
+| `credential` | String | Defines any credential variables required to connect to your chosen endpoint. ML Commons uses **AES/GCM/NoPadding** symmetric encryption with a key length of 32 bytes. When a connection cluster first starts, the key persists in OpenSearch. Therefore, you do not need to manually encrypt the key.
+| `action` | JSON array | Tells the connector what actions to run after a connection to ML Commons has been established. For more information about how to configure actions, see [Actions](#action-settings).
+| `backend_roles` | String | A list of OpenSearch backend roles. For more information about setting up backend roles, see [Assigned backend roles to users]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#assigning-backend-roles-to-users).
```

Comment:
Should it be "Assigning" instead of "Assigned" in the link?

Reply:
Yes.